### PR TITLE
Test suit for FB Type Basic functionality

### LINKED ISSUE #157: Basic FBT Operation Test Module
Part of #153 

Test the fundamental operations related to FBTs

This flow ensures:
-  Manage ports, changing port names, and changing port comment.
-  Delete port, add port, delete data network, add data network.

### PR DESCRIPTION
**Pull request by ShantKhatri**
Fixes #157 
Tested functionalities:

- Add a connection between two pins in FB
- Remove a connection between two pins in FB
- Change data pin comment
- Change event pin comment 
- Change pin data type
- Create a new data Input pin
- Create a new data output pin
- Create a new event input pin
- Create a new event output pin

### CODE REVIEW

File: tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/BasicFBTOperation/BasicFBTOperationTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *
+ *   Prashantkumar Khatri - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.test.ui.BasicFBTOperation;
```
Comment (azoitl) [inline tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/BasicFBTOperation/BasicFBTOperationTest.java]: please don't put each of your test classes in a package that is called like your class. First of all this violates the Java nameing guidelines. Secondly this makes it very hard to navigate your source code. 

Therefore please create one package for your FB type tests, e.g., org.eclipse.fordiac.ide.test.ui.fbtype, and put there all tests belonging to this category types.

**Review by azoitl — CHANGES_REQUESTED** [review]

File: tests/org.eclipse.fordiac.ide.test.ui/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -31,5 +32,5 @@ Require-Bundle: org.eclipse.ui,
  org.eclipse.fordiac.ide.systemmanagement.ui;bundle-version="3.0.0",
  org.eclipse.fordiac.ide.typemanagement;bundle-version="3.0.0",
  org.eclipse.swt,
- org.hamcrest.core,
+ org.hamcrest.library,
```
Comment (azoitl) [inline tests/org.eclipse.fordiac.ide.test.ui/META-INF/MANIFEST.MF]: Could you please extract this fix into an own PR. As you can see in other PRs (e.g., ##269 & #274) this is a generic improvement we need and should therefore be separated.

**Review by azoitl — CHANGES_REQUESTED** [review]

File: tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/BasicFBTOperation/BasicFBTOperationTest.java (hunk 1 above)
Comment (ShantKhatri) [reply tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/BasicFBTOperation/BasicFBTOperationTest.java]: Thank you for your feedback. I acknowledge that it was my mistake to give the package name in the wrong format, which does indeed violate Java naming guidelines. As for the concern about creating new packages for each class, we initially thought it might be necessary to have multiple files under modules, but we now realize that having one file per package is acceptable. Thanks for pointing this out and for your assistance. I have moved the classes accordingly to a single package, as suggested.

File: tests/org.eclipse.fordiac.ide.test.ui/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -33,4 +35,4 @@ Require-Bundle: org.eclipse.ui,
  org.eclipse.swt,
  org.hamcrest.core,
  org.eclipse.draw2d,
- org.hamcrest.library;bundle-version="[1.3.0,2.0.0)"
+ org.hamcrest.library;bundle-version="[1.3.0,2.2.1)"
```
Comment (ShantKhatri) [inline tests/org.eclipse.fordiac.ide.test.ui/META-INF/MANIFEST.MF]: I’m a bit confused about the this change, specifically for:
As version 2.2.1 of org.hamcrest.library is not available yet, and the recommended version for SWTBot 4.2.1 is 2.2.0, I wanted to clarify if it is appropriate to proceed with these changes. Should we align with the recommended version 2.2.0 for org.hamcrest.library, or is there another versioning strategy you’d suggest?

File: plugins/org.eclipse.fordiac.ide.product/org.eclipse.fordiac.ide.product.target [hunk 4]
```diff
@@ -59,13 +59,17 @@
...
 		</location>
+		<location includeMode="planner" includeAllPlatforms="false" includeSource="true" includeConfigurePhase="false" type="InstallableUnit">
+      		<unit id="org.eclipse.swtbot.eclipse.feature.group" version="4.2.1.202406141605"/>
+      		<unit id="org.eclipse.swtbot.eclipse.gef.feature.group" version="4.2.1.202406141605"/>
+      		<unit id="org.eclipse.swtbot.feature.group" version="4.2.1.202406141605"/>
+      		<unit id="org.eclipse.swtbot.generator.feature.feature.group" version="4.2.1.202406141605"/>
+      		<unit id="org.eclipse.swtbot.ide.feature.group" version="4.2.1.202406141605"/>
+      		<unit id="org.eclipse.swtbot.nebula.nattable.finder" version="4.2.1.202406141605"/>
+      		<repository id="SWTBot-3_1" location="https://download.eclipse.org/technology/swtbot/releases/4.2.1/"/>
+    	</location>
```
Comment (ShantKhatri) [inline plugins/org.eclipse.fordiac.ide.product/org.eclipse.fordiac.ide.product.target:72]: I've made changes to update the SWTBot plugins to version 4.2.1. This change was necessary to meet our requirement for using nattable.finder. Based on the recommendation from the SWTBot developers, we need to use version 4.2.1. Therefore, I've separated the SWTBot plugins into their own repository, as the current repository does not include the 4.2.1 version plugins.

File: tests/org.eclipse.fordiac.ide.test.ui/META-INF/MANIFEST.MF (hunk 3 above)
Comment (azoitl) [reply tests/org.eclipse.fordiac.ide.test.ui/META-INF/MANIFEST.MF]: What you currently see is just a workaround that was found in the SWTBot tests of EGit. I think the solution you proposed is a much cleaner one. Therefore I asked you to create an own PR where only your fix to the MANIFEST.MF file is contained.

Comment (ShantKhatri) [reply tests/org.eclipse.fordiac.ide.test.ui/META-INF/MANIFEST.MF]: I have created the new pull request now.

File: tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/Abstract4diacUITests.java [hunk 5]
```diff
@@ -422,6 +466,195 @@ protected static void deleteFBType(final String typeName) {
...
+		for (final SWTBotTreeItem item : parentItem.getItems()) {
+			if (item.getText().contains(typeName)) {
+				projectItem = item;
+				break;
+			}
+		}
+		assertTrue(projectItem.getText().contains(typeName));
+	}
+
+	/**
```
Comment (bwiesmayr) [inline tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/Abstract4diacUITests.java:503]: I think I have a solution for the copied code. Essentially, we need to copy the copyright header if we plan to use the code ourselves. But copying the whole copyright header into this class feels wrong. I therefore suggest to create a new class that has the copied methods and also add the copyright header from SWTBotSiriusHelper

File: tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/PropertySheetHelper.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	private static SWTWorkbenchBot bot = new SWTWorkbenchBot();
+
+	/**
+	 * Select the tab with the name label in the property views.
+	 *
+	 * @param label         Label to find.
+	 * @param propertiesBot the bot corresponding to the property view.
+	 * @return true if the property tab is found, false otherwise
+	 */
+	@SuppressWarnings({ "unchecked" })
```
Comment (bwiesmayr) [inline tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/PropertySheetHelper.java]: what does this suppress warning do?

File: tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/PropertySheetHelper.java [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+import org.eclipse.swtbot.swt.finder.results.BoolResult;
+import org.eclipse.swtbot.swt.finder.utils.SWTUtils;
+import org.eclipse.ui.internal.views.properties.tabbed.view.TabbedPropertyList;
+import org.hamcrest.Matcher;
+import org.hamcrest.Matchers;
+
+public class PropertySheetHelper {
+
+	/**
+	 * SWTWorkbenchBot
```
Comment (bwiesmayr) [inline tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/PropertySheetHelper.java:35]: as discussed earlier, please keep the comment that you also had in the previous version, showing the exact location of this method in SWTBotHelper from Sirius

**Review by bwiesmayr — CHANGES_REQUESTED** [review]

File: tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/PropertySheetHelper.java (hunk 6 above)
Comment (ShantKhatri) [reply tests/org.eclipse.fordiac.ide.test.ui/src/org/eclipse/fordiac/ide/test/ui/PropertySheetHelper.java]: We have utilized methods from the Eclipse Sirius repository and have endeavored to maintain their integrity by not making any modifications to them, so I have not changed anything in the method. I have checked to remove the suppress warning, it does nothing, one warning we are receiving is that is related to we are using "org.eclipse.ui.views.properties.tabbed_3.10.300.v20240424-0956.jar" here, so need to a restriction warning here.

**Review by bwiesmayr — APPROVED** [review]

**Review by bwiesmayr — APPROVED** [review]
